dashboard/modules/reporter/reporter_agent.py

```diff
@@ -603,7 +603,8 @@ async def _perform_iteration(self, publisher):
             await asyncio.sleep(reporter_consts.REPORTER_UPDATE_INTERVAL_MS / 1000)
 
     async def run(self, server):
-        reporter_pb2_grpc.add_ReporterServiceServicer_to_server(self, server)
+        if server:
```

Comment:
Any way to just not run this function instead of adding this logic? Seems a bit ugly

Reply:
I didn't find better way.  The `run` is a common interface of dashboard agent module. We can do anything else in this function, besides starting grpc server. So, if we don't run this function, we could lose some code paths or features.